x/auth/keeper/keeper.go

```diff
@@ -277,3 +269,26 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+// GetBlockProposer returns block proposer
```

Comment:
These three functions, `GetBlockProposer`, `SetBlockProposer`, `RemoveBlockProposer` are unused. I looked into old Heimdall, an they were only used from test case `TestProposer` in file `x/auth/keeper_test.go`.
It looks like test case `TestProposer` hasn't been ported to new Heimdall. If that was intended, then you can just get rid of these three functions, and then you can also get rid of function `ProposerKey()` and var `ProposerKeyPrefix` in file `x/auth/types/keys.go`.

Reply:
They are actually used in `BeginBlocker` and `EndBlocker`, in `app.go` file 
I will port the unit test

Reply:
I added similar [comments](https://github.com/0xPolygon/heimdall-v2/pull/9/files#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7R561) on heimdall-v2 app init PR

Reply:
Resolved (AFAIAC)